deepchem/molnet/load_function/fasta_datasets.py

```diff
@@ -0,0 +1,60 @@
+"""
+FASTA  dataset loader. Based on Clearance loader.
+"""
+import os
+import deepchem as dc
+from deepchem.molnet.load_function.molnet_loader import TransformerGenerator, _MolnetLoader
+from deepchem.data import Dataset
+from deepchem.feat.bert_tokenizer import BertFeaturizer
+from typing import List, Tuple, Optional, Union
+
+FASTA_URL = "https://deepchemdata.s3.us-west-1.amazonaws.com/datasets/uniprot_sprot.fasta"
+
+
+class _FASTALoader(_MolnetLoader):
+
+  def create_dataset(self) -> Dataset:
+    dataset_file = os.path.join(self.data_dir, "sequences.fasta")
+    if not os.path.exists(dataset_file):
+      dc.utils.data_utils.download_url(url=FASTA_URL, dest_dir=self.data_dir)
+    loader = dc.data.FASTALoader(featurizer=self.featurizer, legacy=False)
+    return loader.create_dataset(dataset_file, shard_size=8192)
+
+
+def load_FASTA(

```

Comment:
MoleculeNet loaders are intended for particular datasets (like Uniprot/Swissprot etc). We already have a FASTALoader class for loading general FASTA files